File: server/internal/platform/model/department.go
package model

import (
	"time"

	"gorm.io/gorm"
)

// DepartmentStatus 表示部门状态。
type DepartmentStatus int

const (
	// DepartmentStatusEnabled 表示部门可用。
	DepartmentStatusEnabled DepartmentStatus = 1
	// DepartmentStatusDisabled 表示部门已停用。
	DepartmentStatusDisabled DepartmentStatus = 2
)

// Department 是组织部门表模型。
type Department struct {
	ID           uint             `gorm:"primaryKey" json:"id"`
	ParentID     uint             `gorm:"not null;default:0;index" json:"parent_id"`
	Ancestors    string           `gorm:"size:500;not null;default:''" json:"ancestors"`
	Name         string           `gorm:"size:64;not null" json:"name"`
	Code         string           `gorm:"size:64;not null;uniqueIndex" json:"code"`
	LeaderUserID uint             `gorm:"not null;default:0;index" json:"leader_user_id"`
	Sort         int              `gorm:"not null;default:0" json:"sort"`
	Status       DepartmentStatus `gorm:"type:smallint;not null;default:1" json:"status"`
	Remark       string           `gorm:"size:255;not null;default:''" json:"remark"`
	CreatedAt    time.Time        `json:"created_at"`
	UpdatedAt    time.Time        `json:"updated_at"`
	DeletedAt    gorm.DeletedAt   `gorm:"index" json:"-"`
}

// TableName 固定部门表名。
func (Department) TableName() string {
	return "sys_department"
}

File: admin/src/modules/iam/types/department.ts
// 部门状态常量：启用、禁用
export const DepartmentStatus = {
  Enabled: 1,
  Disabled: 2,
} as const

// 部门状态联合类型
export type DepartmentStatus = (typeof DepartmentStatus)[keyof typeof DepartmentStatus]

// 部门树节点数据结构
export interface DepartmentItem {
  id: number
  parent_id: number
  ancestors: string
  name: string
  code: string
  leader_user_id: number
  sort: number
  status: DepartmentStatus
  remark: string
  created_at: string
  updated_at: string
  children?: DepartmentItem[]
}

// 部门列表查询参数
export interface DepartmentListQuery {
  keyword?: string
  status?: DepartmentStatus | 0
}

// 创建部门的请求载荷
export interface CreateDepartmentPayload {
  parent_id: number
  name: string
  code: string
  leader_user_id: number
  sort: number
  status: DepartmentStatus
  remark: string
}

// 更新部门的请求载荷（与创建载荷相同）
export type UpdateDepartmentPayload = CreateDepartmentPayload

// 更新部门状态的请求载荷
export interface UpdateDepartmentStatusPayload {
  status: DepartmentStatus
}
